### PR TITLE
Add care::copy_n method

### PR DESCRIPTION
**Pull request by adayton1**
Attempting to match std::copy_n

### CODE REVIEW

File: src/care/algorithm_impl.h [hunk 1]
```diff
@@ -979,6 +979,20 @@ CARE_INLINE void fill_n(care::host_device_ptr<T> arr, Size n, const U& val)
...
+ * Author(s) : Alan Dayton
+ * Purpose   : Copies one ManagedArray into another.
+ * ************************************************************************/
+template <class T, class Size, class U>
+CARE_INLINE void copy_n(care::host_device_ptr<T> in, Size n,
+                        care::host_device_ptr<U> out)
+{
+   CARE_STREAM_LOOP(i, 0, n) {
+      out[i] = in[i];
+   } CARE_STREAM_LOOP_END
```
Comment (liu15) [inline src/care/algorithm_impl.h:993]: Since this is inline, we don't need instantiations, right?

**Review by liu15 — APPROVED** [review]

File: src/care/algorithm_impl.h [hunk 2]
```diff
@@ -979,6 +979,20 @@ CARE_INLINE void fill_n(care::host_device_ptr<T> arr, Size n, const U& val)
    } CARE_STREAM_LOOP_END
 }
 
+/************************************************************************
+ * Function  : copy_n
+ * Author(s) : Alan Dayton
+ * Purpose   : Copies one ManagedArray into another.
+ * ************************************************************************/
+template <class T, class Size, class U>
+CARE_INLINE void copy_n(care::host_device_ptr<T> in, Size n,
```
Comment (dtaller) [inline src/care/algorithm_impl.h:988]: Can the in array be made const?

Comment (dtaller) [reply src/care/algorithm_impl.h:988]: Or on the other hand I don't think std::copy_n is const , maybe that change isn't needed?

**Review by dtaller — APPROVED** [review]

File: src/care/algorithm_impl.h (hunk 2 above)
Comment (liu15) [reply src/care/algorithm_impl.h:988]: Const would be nice, if not in the signature, at least before the lambda capture.

File: src/care/algorithm_impl.h (hunk 1 above)
Comment (adayton1) [reply src/care/algorithm_impl.h:993]: I believe that is the case.

File: src/care/algorithm_impl.h (hunk 2 above)
Comment (adayton1) [reply src/care/algorithm_impl.h]: Just made it const.

Comment (adayton1) [reply src/care/algorithm_impl.h]: Looks like I need both a const and non-const overload (I just made the non-const overload call the const overload).

**Comment by adayton1** [comment]
@dtaller, @liu15 I finally addressed your review comments from a long time ago. This is is ready for another review pass.

**Comment by dtaller** [comment]
Looks good to me! Thanks!

**Comment by liu15** [comment]
Looks good to me as well.

**Pull request merged**